investment-app/TRINV.Api/TRINV.Application/Enums/NotificationType.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace TRINV.Application.Enums;
+
+using Shared.Business.Helpers;
+
+public class NotificationType : EnumerationHelper
+{
+    public static readonly NotificationType Email = new NotificationType(0, "Email");
+    public static readonly NotificationType Alert = new NotificationType(1, "Alert");
+    public static readonly NotificationType System = new NotificationType(2, "System");
+
+    public NotificationType(int id, string name) 
+        : base(id, name) {}
+
+    public static string GetMessageById(int id)
+        => id switch
+        {
+            0 => "Default Email Notification",
+            1 => "Default Alert Message",
+            2 => "Default System Message",
+            _ => throw new ArgumentNullException("Unexisting notification type!")
+        };
+
+    public static bool ExistById(int id)
```

Comment:
Mhm it's not very clear approach. Every time when we add new notification type. We would need to come back and adjust this method.

Implemeneted in different way. For example. 
You can add GetAll method in  the abstrac class EnumerationHelper

Something like that -> 

`
public static IEnumerable<T> GetAll<T>() where T : EnumerationHelper =>
		typeof(T).GetFields(BindingFlags.Public |
							BindingFlags.Static |
							BindingFlags.DeclaredOnly)
				 .Select(f => f.GetValue(null))
				 .Cast<T>();

`

And use it something like that 
EnumerationHelper.GetAll<NotificationType>().Any(x => x.Id == request.NotificationType)

Reply:
Done.